### PR TITLE
kadane's Algorithm in C

### PR DESCRIPTION
**Pull request by raksha009**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

Max subarray sum implemented using Kadane's Algorithm in C.

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [X] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
ISSUE: #178 

Plz review my Pr @Kajol-Kumari  @harsh-9in

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@raksha009 rename the file to `maxSubArraySum` since Kadane's Algorithm is more general algorithm and can be used in other problems but in your code, you have implemented `maxSubArraySum` which uses Kadane's Algorithm so renaming the file to `maxSubArraySum.c` will look better. Also, add time and space complexity at the end of the file.

**Comment by raksha009** [comment]
Done. Plz review and merge my pr @iamrajiv

File: C/cp/maxSubArraySum.c [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	int arr[n];
+	printf("Enter elemets of array \n");
+	for (int i = 0; i < n; i++ ){
+	    scanf(" %d ", &arr[i]);
+	}
+	
+	//KADANE'S ALGORITHM
+	int max_sum  = arr[0];
+	int curr_max = arr[0];
+	for (int i = 1; i < n; i++ ){
```
Comment (SKAUL05) [inline C/cp/maxSubArraySum.c]: Change `i` to some suitable variable names.

File: C/cp/maxSubArraySum.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#include <stdlib.h>
+#include <math.h>
+#define MIN(a,b) (((a)<(b))?(a):(b))
+#define MAX(a,b) (((a)>(b))?(a):(b))
+
+
+int main(void) {
+    
+	// Aceepting an array 
+	int n;
```
Comment (SKAUL05) [inline C/cp/maxSubArraySum.c]: Change all alphabets to suitable variable names

**Review by SKAUL05 — CHANGES_REQUESTED** [review]

File: C/cp/maxSubArraySum.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#include <math.h>
+#define MIN(a,b) (((a)<(b))?(a):(b))
+#define MAX(a,b) (((a)>(b))?(a):(b))
+
+
+int main(void) {
+    
+	// Aceepting an array 
+	int size;
+	printf("Enter number of elemets in array \n");
```
Comment (Kajol-Kumari) [inline C/cp/maxSubArraySum.c]: please change `printf("Enter number of elemets in array \n");` to `printf("Enter the size of the array \n");`

File: C/cp/maxSubArraySum.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	scanf (" %d ", &size);
+	
+	int arr[size];
+	printf("Enter elemets of array \n");
```
Comment (Kajol-Kumari) [inline C/cp/maxSubArraySum.c]: please correct the typo here `elemets` to  `elements`

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @raksha009 I am not sure why this is happening but when i ran your program it's taking input in the way i have attached below and also take a look at the answer of the code snippet i have pasted.
Can you please look at it, why is it happening?

```
Enter number of elemets in array                                                                                                
1                                                                                                                               
6                                                                                                                               
Enter elemets of array                                                                                                          
1                                                                                                                               
Maximum sum of subarray is 6
```

**Comment by raksha009** [comment]
Done the changes @Kajol-Kumari

File: C/cp/maxSubArraySum.c [hunk 5]
```diff
@@ -0,0 +1,52 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#define MIN(a,b) (((a)<(b))?(a):(b))
+#define MAX(a,b) (((a)>(b))?(a):(b))
+
```
Comment (iamrajiv) [inline C/cp/maxSubArraySum.c:4]: @raksha009 remove the unnecessary line break

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@raksha009 add time and space complexity

File: C/cp/maxSubArraySum.c [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	}
+	
+	printf("Maximum sum of subarray is %d \n", max_sum);
+	
+	return 0;
+}
+
+/*
+TIME COMPLEXITY : O(N) where n is number of elements in array
+SPACE COMPLEXITY : o(1) as no new space is allocated
```
Comment (iamrajiv) [inline C/cp/maxSubArraySum.c]: @raksha009 change `o(1)` to `O(1)`

**Review by iamrajiv — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]

**Review by SKAUL05 — APPROVED** [review]

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :)

**Comment by raksha009** [comment]
Plz merge my PR @Kajol-Kumari @SKAUL05 @iamrajiv

**Comment by iamrajiv** [comment]
@all-contributors please add @raksha009 for code

**Pull request merged**